src/sinks/datadog/logs.rs

```diff
@@ -70,13 +74,20 @@ impl GenerateConfig for DatadogLogsConfig {
 }
 
 impl DatadogLogsConfig {
-    fn get_endpoint(&self) -> &str {
-        self.endpoint
-            .as_deref()
-            .unwrap_or_else(|| match self.region {
-                Some(super::Region::Eu) => "https://http-intake.logs.datadoghq.eu",
-                None | Some(super::Region::Us) => "https://http-intake.logs.datadoghq.com",
-            })
+    fn get_uri(&self) -> String {
+        self.endpoint.clone().unwrap_or_else(|| {
+            self.site
+                .as_ref()
+                .map(|s| format!("https://http-intake.logs.{}/v1/input", s))
+                .unwrap_or_else(|| match self.region {
+                    Some(super::Region::Eu) => {
+                        "https://http-intake.logs.datadoghq.eu/v1/input".to_string()
+                    }
+                    None | Some(super::Region::Us) => {
+                        "https://http-intake.logs.datadoghq.com/v1/input".to_string()
+                    }
+                })
+        })
```

Comment:
Nit: I think this would be a bit easier to follow without the nesting. Something like:

```rust
self.endpoint.clone()
  .or_else(|| self.site.as_ref().map(|s| format!("https://http-intake.logs.{}/v1/input", s))
  .unwrap_or_else( || match self.region {
                    Some(super::Region::Eu) => {
                        "https://http-intake.logs.datadoghq.eu/v1/input".to_string()
                    }
                    None | Some(super::Region::Us) => {
                        "https://http-intake.logs.datadoghq.com/v1/input".to_string()
                    }
                })
```

This reads clearer to me that we use the endpoint, if it exists, or we use the site, or we use the region.